### DISCUSSION TITLE
How does Super Editor handle serialisation deserialisation? What data format is used?

### DISCUSSION THREAD

**Discussion by adrian-moisa** [Flutter-Bounty-Hunters/super_editor#548] [General]
I've been using flutter Quill a lot lately. It handles the data in the delta format, which I find super handy for many reasons. I'm able using delta attributes to store custom attributes interpreted by my app and to react to them either in Quill or in the app itself. I'm curios to learn about SuperEditor how is the document serialisation handled? I could not find a clear answer in the docs.

Then I watched the video about SU 2.0. You mentioned coop editing. How is this going to be implemented? How do you plan to transfer the deltas? Cheers!

**Comment by bk-one** [comment]
Superlist is planning to release a package for (de-)serialization to Delta. We're already using it internally, but it's not yet stable enough to release it as a package. SuperEditor will always be agnostic to the format used, but add-ons should allow opinionated ways for serialization.

**Reply by adrian-moisa** [reply]
What about the coop editing? how will this work, any technical details? Cheers!

**Reply by bk-one** [reply]
The Delta format allows coop editing to a certain extend. To build a fully working Google Docs like experience, a lot of factors have to be taken into account that can't be easily abstracted. It depends a lot on the use cases for your specific application. You can try out Superlist and see how much the real-time collaboration fits your need and we can talk through the details. Don't expect this package to be a one-stop solution to build real-time, collaborative editing experiences. It will provide all the tools and components (e.g. we're working on displaying multiple carets in a document right now), but assembling them in a way that fits your need is up to you, especially as there's always a backend component involved which is definitely out of scope for SuperEditor.

**Reply by adrian-moisa** [reply]
Oh I see, so basically super editor provides the rendering tooling needed for multiple coop parties to join live (such as floating cursor, parsing diff objects) but the transit of this information is a duty that each developer has to fulfil by whatever tech stack they are using.

**Comment by peter100u** [comment]
Thank you very much for this great project. I have already used super_editor for editing notes in my project. Now, how can I upload the content edited by super_editor to Firebase, in what format? Thank you for your answer, after all, this is only one step away from publishing
